### ISSUE TITLE
Export/Import of tab control with a shared picture in the caption does not work

### ISSUE THREAD

**Issue by Ik-12** [joyfullservice/msaccess-vcs-integration#170]
I have a form with a tab control and I would like to use a shared picture in the caption of
tab pages. However, this does not seem to work very along with the VCS system, because
after importing the "Use theme" property of the tab control changes to "No" which
causes that Access cannot find the shared picture.

To reproduce the issue, please export and import the code in the attached example 
and try to open the "Form1" (the zip contains a git repo so you can also easily see 
the changes in `Form1.bas` after export). 

According to my testing the problem is in the `SanitizeFile` function. Removing **any** of the lines 
removed by "Aggresive Sanitize" or "Strip out Publish Option" options from the file produced by
`Application.SaveAsText` breaks handling of `UseTheme` property.  In addition, also the "Checksum =" 
line seems to be necessary.

Any ideas how to fix this? Totally disabling sanitizing does not seem a good idea as 
this would lead in numerous irrelevant changes every time the form is edited.

(Tested with the `dev` branch, Access 2016 64-bit, Windows 10)

[TabWithPicture.accdb.src.zip](https://github.com/joyfullservice/msaccess-vcs-integration/files/5986722/TabWithPicture.accdb.src.zip)

**Comment by hecon5** [comment]
Does the checksum have to be valid? I was looking at that, and the checksum appears to be based on the output file, so since it's being sanitized, this may break things.

**Comment by hecon5** [comment]
**TL/DR**
 After making the fix you proposed, this appears to be fixed (and using my dev branch which has more error handling from PR joyfullservice#162 and will be making its way into dev soon hopefully).

**Long, Drawn Out and Over-Verbose Version:**
I'm also on the same version of Access, I originally got error 2128, and the error file was generated for Form1 in your example. This validates your bug find, as the `Found` value is the remainder of the `GUID` string that was and is no more.

```
Error encountered at line 19.
Expected: 'End'.  Found: 0x7499ccbcd01efc4dade876d6fe35d91f.
```

I made the fix to the sanitize function you mentioned, and the form seems to import fine with `Strip Out publish` set to `True` and `Aggressive Sanitize` FALSE, 

I then (just for giggles) turned on Aggressive sanitize again, and it still appears to import fine, I get a plus symbol on the Page1 Tab. 

After getting this to work, I deleted my (now modified) copy of your source, I then re-extracted your original zip, did a full export with all the standard options (`AgressiveSanitize` and `StripOutPublish` both `TRUE`), 

Original:
![image](https://user-images.githubusercontent.com/54177882/108069664-9e829b00-7031-11eb-812d-5a9b9f944e78.png)

After Rebuild:
![image](https://user-images.githubusercontent.com/54177882/108069577-84e15380-7031-11eb-90bf-6afabef26aeb.png)

**Comment by hecon5** [comment]
When you changed settings, did you perchance use `export full source` the first time? 

I think the bug may be caused by a setting change that has not yet been exported (for instance, if you exported with `aggressive sanitize` `FALSE` and exported, then changed it to`TRUE` re-exported, then re-imported, unless you made a change to `form1`, the file would still be broken. 

@joyfullservice: it appears that after making a setting change the next export should export all the source to ensure exports match the current settings.

**Comment by Ik-12** [comment]
> 
> 
> Does the checksum have to be valid? I was looking at that, and the checksum appears to be based on the output file, so since it's being sanitized, this may break things.

Apparently yes. At least when I simply export the form with Application.SaveAsText, change checksum with a text editor
and try to load it with Application.LoadFromText, I get the "Access can't open the file 'Page1'" error. If I do not modify the 
checksum this works.

**Comment by Ik-12** [comment]
> 
> 
> When you changed settings, did you perchance use `export full source` the first time?
> 
> I think the bug may be caused by a setting change that has not yet been exported (for instance, if you exported with `aggressive sanitize` `FALSE` and exported, then changed it to`TRUE` re-exported, then re-imported, unless you made a change to `form1`, the file would still be broken.
> 
> @joyfullservice: it appears that after making a setting change the next export should export all the source to ensure exports match the current settings.

Initially I used "Export All Source" with default options. It seems that after changing settings "Full export"
option is not needed. The form is exported automatically although it has not been edited. Anyway, every time
I changed any settings, I verified from `git diff` that the form export file (`Form1.bas`) actually changed.

I will test with your version when it has been merged into the `dev` branch but according the screenshots it works 
(which would mean that my diagnose about the problem was wrong?) .

**Comment by hecon5** [comment]
I would bet that it was a freak export error, then you changed settings, and the guid block was broken, which broke the import, and sent you to the abyss :). I think the aggressive sanitize bug you found was definitely the cause of anything after the initial export.

With the improved error handling / logging, troubleshooting should improve :)

**Comment by joyfullservice** [comment]
I am able to export and build your sample file with no errors. After build, the shared image is shown in the tab control. The form export is identical to the one in your sample. Looking at the source in `Form1.bas`, I don't see any broken or missing blocks...

![image](https://user-images.githubusercontent.com/7254992/108092197-12c53a80-7042-11eb-8722-38e7527e6da4.png)

![image](https://user-images.githubusercontent.com/7254992/108092264-253f7400-7042-11eb-968a-e7a2eee20822.png)

I am using Access 2010, but I am not seeing any reproducible issue here...

**Comment by hecon5** [comment]
Turn off `Aggressive Sanitize` and it will throw an error.

**Comment by Ik-12** [comment]
It seems evident that the issue is version specific and more an Access "feature" 
than a bug in the addin.

As I already wrote, I can easily reproduce the problem with using only Application.SaveAsText/Application.LoadFromText and removing e.g. 
the "Checksum=" line using text editor.

I will test tomorrow at office with different platforms and Access versions. 
This is not the only thing that behaves strangely with Access on my laptop...

**Comment by Ik-12** [comment]
I tested with two other computers:

1. Access 2019, Windows 10: works ok
2. Access 2016, Windows 10: does not work

I also found an old issue for this problem: https://github.com/msaccess-vcs-integration/msaccess-vcs-integration/issues/41

I verified all Access settings on the computers and didn't find any differences. Based on this evidence, my guess is that this is a bug affecting Access versions 2013 and 2016.  @hecon5: which Access version you are using?

**Comment by hecon5** [comment]
I was using 2013 32bit until recently, now am on 2016 64 bit.

**Comment by hecon5** [comment]
Update: my machine is now apparently on 365PP, so I lied; sorry @Ik-12!

**Comment by joyfullservice** [comment]
Just to follow up here, is this still an outstanding issue? If it only affects a couple specific versions of Access, do we want to close this out as an Access bug that may eventually get fixed, or is it important enough to try to develop a workaround in the add-in?

**Comment by hecon5** [comment]
From my perspective, I'm not even sure how we could go about fixing this, as it looks like an issue with the import in access itself, and the fix would be to roll our own importer (big lift). Unlike with queries/tables, where there are only a finite number of fields and they're all text based (ish), Forms/reports could have a gargantuan number of things we'd need to accommodate. And since it's been fixed in a "newer" version of Access, I'd suggest we not try and tackle it.
